DigitalLearningSolutions.Web/Controllers/TrackingSystem/CentreConfiguration/RegistrationPromptsController.cs

```diff
@@ -103,88 +112,137 @@ public IActionResult AddRegistrationPromptSelectPrompt(AddRegistrationPromptSele
+        }
+
+        [HttpGet]
+        [Route("Add/ConfigureAnswers")]
+        [ServiceFilter(typeof(RedirectEmptySessionData<AddRegistrationPromptData>))]
+        public IActionResult AddRegistrationPromptConfigureAnswers()
+        {
             var data = TempData.Peek<AddRegistrationPromptData>()!;
-            data.SelectPromptViewModel = model;
-            TempData.Set(data);
+            var viewModel = data.ConfigureAnswersViewModel;
 
-            // TODO: HEEDLS-453 - redirect to next page
-            return RedirectToAction("Index");
+            return View(viewModel);
+        }
+
+        [HttpPost]
+        [Route("Add/ConfigureAnswers")]
+        [ServiceFilter(typeof(RedirectEmptySessionData<AddRegistrationPromptData>))]
+        public IActionResult AddRegistrationPromptConfigureAnswers
+            (RegistrationPromptAnswersViewModel model, string action)
+        {
+            if (action.StartsWith(DeleteString) && TryGetAnswerIndexFromDeleteAction(action, out var index))
+            {
+                return RegistrationPromptAnswersPostRemovePrompt(model, index, true);
+            }
+
+            return action switch
+            {
+                NextString => AddRegistrationPromptConfigureAnswersPostNext(model),
+                AddPromptString => RegistrationPromptAnswersPostAddPrompt(model, true),
+                _ => RedirectToAction("Error", "LearningSolutions")
+            };
         }
 
-        private bool TryGetAnswerIndexFromAction(string action, out int index)
+        private IActionResult EditRegistrationPromptPostSave(EditRegistrationPromptViewModel model)
         {
-            return int.TryParse(action.Remove(0, 6), out index);
+            IgnoreAddNewAnswerValidation();
+
+            if (!ModelState.IsValid)
+            {
+                return View(model);
+            }
+
+            customPromptsService.UpdateCustomPromptForCentre
+                (User.GetCentreId(), model.PromptNumber, model.Mandatory, model.OptionsString);
+
+            return RedirectToAction("Index");
         }
 
-        private IActionResult EditRegistrationPromptPostAddPrompt(EditRegistrationPromptViewModel model)
+        private IActionResult RegistrationPromptAnswersPostAddPrompt
+            (RegistrationPromptAnswersViewModel model, bool saveToTempData = false)
         {
             if (!ModelState.IsValid)
             {
-                model.Options = NewlineSeparatedStringListHelper.SplitNewlineSeparatedList(model.OptionsString);
                 return View(model);
             }
 
-            var (optionsString, options) =
+            var optionsString =
                 NewlineSeparatedStringListHelper.AddStringToNewlineSeparatedList(model.OptionsString, model.Answer!);
 
             if (optionsString.Length > 4000)
             {
-                AddTotalStringLengthModelError(model);
+                AddTotalStringLengthRegistrationPromptAnswersViewModelError(model);

```

Comment:
Maybe name this to something like SetConfiguredAnswersTooLongError or something similar?